### PR TITLE
Preliminary FreeBSD support.

### PR DESCRIPTION
**Pull request by jsm222**
Encountered issues so far:

 open /dev/snd/seq failed:
 The score starts zoomed in, so one has to zoom
 out with the ctrl plus mouse wheel

### CODE REVIEW

**Comment by cbjeukendrup** [comment]
> The score starts zoomed in, so one has to zoom
> out with the ctrl plus mouse wheel

That sounds similar to the problem that https://github.com/musescore/MuseScore/pull/17759 is solving. Might be worth trying that out!

File: thirdparty/dr_libs/dr_mp3.h [hunk 1]
```diff
@@ -3151,9 +3151,11 @@ static drmp3_result drmp3_result_from_errno(int e)
     #ifdef ENOTUNIQ
         case ENOTUNIQ: return DRMP3_NOT_UNIQUE;
     #endif
+#ifndef __FreeBSD__
     #ifdef EBADFD
         case EBADFD: return DRMP3_ERROR;
     #endif
+#endif
```
Comment (igorkorsukov) [inline thirdparty/dr_libs/dr_mp3.h:3158]: Can you please explain why this needs to be added?

File: thirdparty/fluidsynth/fluidsynth-2.1.4/src/external/portable_endian.h [hunk 2]
```diff
@@ -52,9 +52,9 @@
 
 #	include <sys/endian.h>
 
-#elif defined(__NetBSD__) || defined(__FreeBSD__) || defined(__DragonFly__)
+#elif defined(__NetBSD__) || defined(__DragonFly__)
 
-#	include <sys/endian.h>
+#x	include <sys/endian.h>
```
Comment (igorkorsukov) [inline thirdparty/fluidsynth/fluidsynth-2.1.4/src/external/portable_endian.h:57]: It looks like the `x` symbol is superfluous here

Comment (jsm222) [reply thirdparty/fluidsynth/fluidsynth-2.1.4/src/external/portable_endian.h:57]: Yes that was a typo. I have refactored the portable_endian.h a litte

**Comment by igorkorsukov** [comment]
@jsm222 please update with the master to make sure the vtests fail is not related to this PR

**Comment by Jojo-Schmitz** [comment]
Don't use merge, but something like `git pull --rebase upstream master; git push -f`

**Comment by jsm222** [comment]
> Don't use merge, but something like `git pull --rebase upstream master; git push -f`

I guess we can do a squash, when/if this work is accepted.

**Comment by jsm222** [comment]
> > The score starts zoomed in, so one has to zoom
> > out with the ctrl plus mouse wheel
> 
> That sounds similar to the problem that #17759 is solving. Might be worth trying that out!

My edid was broken and qt reported 3048 as DPI when it should be 96. The dimension of the screen were not correct in xrandr --verbose. (I have not checked qt internals) Specifying -D 96 *without* your patch worked.

**Pull request merged**